libs/ml/include/ml/ops/divide.hpp

```diff
@@ -44,39 +45,81 @@ class Divide : public fetch::ml::Ops<T>
   virtual void Forward(VecTensorType const &inputs, ArrayType &output)
   {
     assert(inputs.size() == 2);
-    assert(inputs.at(0).get().shape() == inputs.at(1).get().shape());
     assert(inputs.at(0).get().shape() == output.shape());
 
-    fetch::math::Divide(inputs.at(0).get(), inputs.at(1).get(), output);
+    if ((inputs.at(0).get().shape() == inputs.at(1).get().shape()) ||
+        (inputs.at(1).get().size() > 1))
+    {  // array / array
+      fetch::math::Divide(inputs.at(0).get(), inputs.at(1).get(), output);
+    }
+    else
+    {  // array / scalar
+      fetch::math::Divide(inputs.at(0).get(), *(inputs.at(1).get().cbegin()), output);
+    }
   }
 
   /**
-   * f'(x)=(1/y)*err
-   * f'(y)=-(x/(y^2))*err
+   * f'(a)=(1/b)*err
+   * f'(b)=-(a/(b^2))*err
    */
   virtual std::vector<ArrayType> Backward(VecTensorType const &inputs,
                                           ArrayType const &    error_signal)
   {
     ArrayType return_signal_1(inputs.at(0).get().shape());
-    ArrayType return_signal_2(return_signal_1.shape());
+    ArrayType return_signal_2(inputs.at(1).get().shape());
+    return_signal_2.Fill(static_cast<DataType>(0));
```

Comment:
this is unnecessary - When you create a Tensor it should be filled with 0 by default